nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -339,6 +446,81 @@ WaypointFollower::dynamicParametersCallback(std::vector<rclcpp::Parameter> param
   return result;
 }
 
+std::vector<geometry_msgs::msg::PoseStamped>
+WaypointFollower::convertGPSPoses2MapPoses(
+  const std::vector<geographic_msgs::msg::GeoPose> & gps_poses,
+  const rclcpp_lifecycle::LifecycleNode::SharedPtr & parent_node,
+  const std::unique_ptr<nav2_util::ServiceClient<robot_localization::srv::FromLL>> & fromll_client)
+{
+  RCLCPP_INFO(parent_node->get_logger(), "Converting GPS waypoints to %s Frame..", 
+        global_frame_id_.c_str());
+
+  std::vector<geometry_msgs::msg::PoseStamped> poses_in_map_frame_vector;
+  int waypoint_index = 0;
+  for (auto && curr_geopose : gps_poses) {
+    auto request = std::make_shared<robot_localization::srv::FromLL::Request>();
+    auto response = std::make_shared<robot_localization::srv::FromLL::Response>();
+    request->ll_point.latitude = curr_geopose.position.latitude;
+    request->ll_point.longitude = curr_geopose.position.longitude;
+    request->ll_point.altitude = curr_geopose.position.altitude;
+
+    fromll_client->wait_for_service((std::chrono::seconds(1)));
+    if (!fromll_client->invoke(request, response)) {
+      RCLCPP_ERROR(
+        parent_node->get_logger(),
+        "fromLL service of robot_localization could not convert %i th GPS waypoint to"
+        "%s frame, going to skip this point!"
+        "Make sure you have run navsat_transform_node of robot_localization", waypoint_index, global_frame_id_.c_str());
+      if (stop_on_failure_) {
+        RCLCPP_ERROR(
+          parent_node->get_logger(),
+          "Conversion of %i th GPS waypoint to"
+          "%s frame failed and stop_on_failure is set to true"
+          "Not going to execute any of waypoints, exiting with failure!", waypoint_index, global_frame_id_.c_str());
+        return std::vector<geometry_msgs::msg::PoseStamped>();
+      }
+      continue;
+    } else {
+      // fromll_client converted the point from LL to Map frames
+      // but it actually did not consider the possible yaw offset between utm and map frames ;
+      // "https://github.com/cra-ros-pkg/robot_localization/blob/
+      // 79162b2ac53a112c51d23859c499e8438cf9938e/src/navsat_transform.cpp#L394"
+      // see above link on how they set rotation between UTM and
+      // Map to Identity , where actually it might not be
```

Comment:
@SteveMacenski, there was an [ongoing discussion](https://github.com/ros-planning/navigation2/pull/2111/files#r564013242) about this in #2111 about whether an issue should be opened on `robot_localization` to address this. Should we review this there?

Also, do you think it may be a good idea to modify the `fromLL` service or include a new one to accept an array of pairs of coordinates and return an array of poses? It may be a common necessity IMO to call the service over multiple points at once, which may be simplified by the inclusion of such a service.

Reply:
Sure, I think that would be valuable and should be easy for Tom and I to review in RL. As well on the RL ticket for orientation, though you made it seem like that actually wasn't an issue in the call?

Reply:
Hmm, after reviewing the code more in detail its actually true that RL does not account for this possible offset between the `map` and `utm` frames, however I'm not sure how significant this offset can be to actually cause an important difference in the result of the conversion as it is my understanding that both `utm` and `map` frames need to follow the ENU convention, and thus the X axis of both should be aligned with the east. 

However I may be wrong about this (I'm not an expert in geography, if I'm wrong please educate me) , so IMO its worth it to post this question on RL since the maintainers probably know much more about this than I do. 

Anyway, I think it won't hurt to keep this rotation logic in the code while this is cleared out, because if the frames are indeed aligned it should not change the waypoint coordinates in the map frame.

Reply:
I will ask both questions on RL. Meanwhile i think this can be merged as it is now.

Reply:
There's no convention in the orientation of the `map` frame, it is completely arbitrary based on the starting pose of the robot when the map is built. There's no global alignment standard for the map

Reply:
Im aware this is the case with nav2, however i think that RL does follow the `ENU` standard as stated [here](https://github.com/cra-ros-pkg/robot_localization/blob/ros2/src/navsat_transform.cpp#L266). Since the code assumes RL is being used (it relies on the `/fromLL` service) i think it would be reasonable to assume that for the excecution of the GPS action server the map frame follows the `ENU` convention. Do you think this may have any harmful effects?

Reply:
I don't honestly know, is that stated somewhere that RL does that for the conversion? I'm just unfamiliar with that bit, that could very well be true, I honestly just don't know the answer.

Reply:
Hi, just sorted this out with the help of Tom. You and him are right and this offset compensation is harmful. RL produces the correct output even when IMU data does not follow the ENU convention and `map` and `utm` frames are not aligned.
Resolving this now.

Reply:
I know this is marked as resolved, but I've got a little real world data in this [question/answer](https://answers.ros.org/question/410550/declination-usage-in-navsat_transform/). The image in this question visualizes what happens when the erroneous rotation was used on a real bot.